Enable floating promises check everywhere (fix issues)

mcortesi opened the pull request:
### Description

Removes floagint promises on all packages that inherit from tslint
parent config.

Only in `mobile` package, the config is overrided as a "warning".

Also: updates tslint & tslint rule set to their last versions.

### Tested

`yarn lint`

### Other changes

Upgrade tslint

### Related issues

- Related to #238 
- Fixes #1080

cmcewen commented on packages/contractkit/src/address-registry.ts:
```diff
@@ -22,8 +21,8 @@ export class AddressRegistry {
       debug('Fetching address from Registry for %s', contract)
-      const hash = Web3.utils.soliditySha3({ type: 'string', value: contract })
-      const address = await this.registry.methods.getAddressFor(hash).call()
+      // const hash = Web3.utils.soliditySha3({ type: 'string', value: contract })
```
why comment this out?

mcortesi replied:
Thanks for catching this.. my commit included some other work that should not be here.

Now that's fixed

cmcewen commented on packages/verifier/src/components/HomeScreen/HomeScreen.tsx:
```diff
@@ -86,20 +86,24 @@ class HomeScreen extends React.Component<Props, State> {
-    NetInfo.addEventListener('connectionChange', this.handleNetworkStatusChange)
+    try {
+      await this.getVerifierStatus()
+      NetInfo.addEventListener('connectionChange', this.handleNetworkStatusChange)
```
I think this should probably go above anything we await

cmcewen commented on packages/verifier/src/components/HomeScreen/HomeScreen.tsx:
```diff
@@ -86,20 +86,24 @@ class HomeScreen extends React.Component<Props, State> {
 
-    if (!this.props.isVerifying && isVerifyingDisabledLongTime(this.props.verifyingOffAt)) {
-      setTimeout(() => this.showVerifyingOffLongMessage(), 2000)
+      if (!this.props.isVerifying && isVerifyingDisabledLongTime(this.props.verifyingOffAt)) {
```
also move this above await probably

mcortesi committed:
Enable floating promises check everywhere (fix issues)

Removes floagint promises on all packages that inherit from tslint
parent config.

Only in mobile package, the config is overrided as a "warning".

Also: updates tslint & tslint rule set to their last versions.

jmrossy reviewed (approved):
Very nice!

Commit:
Merge branch 'master' into mc/floating-promises

Commit:
Merge branch 'master' into mc/floating-promises

Commit:
Merge refs/heads/master into mc/floating-promises
